common/src/GtinHandler.php

```diff
@@ -0,0 +1,54 @@
+<?php
+namespace Valued\WordPress;
+
+class GtinHandler {
+	private $supported_plugings = [
+		'woocommerce-product-feeds/woocommerce-gpf.php' => [
+			'callback' => 'handleGpf',
+			'key' => 'gtin',
+		],
+		'customer-reviews-woocommerce/ivole.php' => [
+			'callback' => 'getGtinFromMeta',
+			'key' => '_cr_gtin',
+		],
+		'product-gtin-ean-upc-isbn-for-woocommerce/product-gtin-ean-upc-isbn-for-woocommerce.php' => [
+			'callback' => 'getGtinFromMeta',
+			'key' => '_wpm_gtin_code',
+		],
+		'woo-product-feed-pro/woocommerce-sea.php' => [
+			'callback' => 'getGtinFromMeta',
+			'key' => '_woosea_gtin',
+		],
+	];
+	private $product;
+
+	public function __construct(\WC_Product $product = null) {
+		$this->product = $product;
+	}
+
+	public function getActivePlugin(): ?array {
+		$plugins = get_plugins();
+		foreach (get_option('active_plugins') as $p) {
+			if (isset($plugins[$p]) && isset($this->supported_plugings[$p])) {
+				return $this->supported_plugings[$p];
+			}
+		}
+		return null;
+	}
+
+	public function getGtin(): ?string {
+		$active_plugin = $this->getActivePlugin();
+		if ($active_plugin) {
+			return call_user_func([$this, $active_plugin['callback']], $active_plugin['key']);
```

Comment:
I would avoid using `call_user_func` unless its really necessary.  In this case you might use closures, or just write an `if` tree.  By using `call_user_func` and strings instead of function names, you lose the IDE's (or [Phan](https://github.com/phan/phan)'s) static analysis that warns you about non-existent methods, etc.

~~~
if ($this->hasPlugin('woocommerce-product-feeds/woocommerce-gpf.php')) {
  return $this->handleGpf('gtin');
}
if ($this->hasPlugin('...')) {
  return $this->getGtinFromMeta('...');
}
// etc.
~~~

Because the cases where it is a simple meta value really are identical, those could go in an array.  Point is to avoid function references in the array that can't be checked by the IDE:

~~~
const GTIN_META_KEYS = [
  'plugin-name/plugin-file.php' => 'meta_key',
  ...
];
~~~

~~~
foreach (self::GTIN_META_KEYS as $plugin => $key) {
  if ($this->hasPlugin($plugin)) {
    return $this->getGtinFromMeta($key);
  }
}
~~~